### DISCUSSION TITLE
miss’throat.area' and miss'pore.area'

### DISCUSSION THREAD

**Discussion by Xiaozizhuo** [PMEAL/OpenPNM#2343] [Q&A] [answered]
Dear Professor Jeff, I am a beginner in the construction of porous media models. When I was learning the toolkit OpenPNM developed by you, I ran into the following problem when running the code on the tutorial, and I don't know how to solve 

For example, it will give the following error
![error](https://user-images.githubusercontent.com/98071787/158955552-422d8826-c783-4c19-b727-13c6b0672287.jpg)

![error1](https://user-images.githubusercontent.com/98071787/158956073-3cd0c9bb-ee10-4e60-ab6b-733861326bff.png)

I guess it is because the version does not match the example of the online tutorial after the version is updated?
Sincerely ask professor jeff to answer my doubts

**Comment by ma-sadeghi** [comment, marked as answer]
We've renamed `pore.area` to `pore.cross_sectional_area` because `pore.area` was vague. Similarly, for `throat.area`. We haven't officially released v3 though. Which version are you using? Can you paste what you get after running `openpnm.__version__`?

**Reply by Xiaozizhuo** [reply]
Thanks for your warm help
My version is 2.8.2.These errors are all getting from running the examples on Github
![error1](https://user-images.githubusercontent.com/98071787/159110591-da477842-e10e-4dbd-8725-82b8c83d7ecb.png)
![error2](https://user-images.githubusercontent.com/98071787/159110595-66a8307b-df71-44f0-81d4-1cc13f84d4d8.png)
![error3](https://user-images.githubusercontent.com/98071787/159110598-a95c9c09-fd7e-4ae7-944c-d5d195503b24.jpg)

**Reply by ma-sadeghi** [reply]
Uninstall your current installation (`pip uninstall openpnm`), and install the development version using `pip install git+https://github.com/PMEAL/OpenPNM.git@dev`.

**Reply by Xiaozizhuo** [reply]
thank you for your reply.I followed your method and got the following error.
![error1](https://user-images.githubusercontent.com/98071787/159148329-1f142329-92b7-4eeb-b67f-58285c23a7aa.jpg)
After I modified it according to the online scheme, the same error still occurs
![error2](https://user-images.githubusercontent.com/98071787/159148343-07de4857-dd6f-4639-9669-7ffa3929b860.jpg)

**Reply by ma-sadeghi** [reply]
It seems that there's something wrong with your internet connection, maybe you're connected via a proxy server? See [here](https://stackoverflow.com/a/18356572). I just tried on a fresh environment on my Linux machine and it worked.

**Reply by ma-sadeghi** [reply]
Also, see [here](https://blog.victorsilva.com.uy/failed-to-connect-to-github/), apparently they had a similar problem.

**Reply by Xiaozizhuo** [reply]
Thanks for your answer, the problem has been solved

**Reply by GaoC-en** [reply]
@Xiaozizhuo  Dear author, I have met the same question with you. And I have read through the answers above and updated my version following the Sadeghi's answer. But it still runs an error like yours. Could you please tell me how to deal with this problem? Thanks a lot for your help.  I am the postgraduate student in Tongji Univ. and major in civil engineering. If it is possible, could we talk about the problems of using OpenPNM usually?

**Reply by Xiaozizhuo** [reply]
可以的，我的联系方式是QQ1511733551 我也存有很多疑惑，最近在做本科毕业设计目前陷入了窘境

**Reply by minzhangut** [reply]
Hello @Xiaozizhuo I am wondering whether you could post here how did you solve this problem. I also have the same issue. Thanks!
